### ISSUE TITLE
Migrate to Puppeteer

### ISSUE THREAD

**Issue by astefanutti** [astefanutti/decktape#100]
From early testing of [Puppeteer](https://github.com/GoogleChrome/puppeteer), it seems it'd be possible to migrate to it to leverage headless Chrome.

1. Export engine:
    - [x] Force the `screen` CSS media type for presentation frameworks that define some `print` styles
        - See GoogleChrome/puppeteer#312
        - In the meantime, `print` CSS rules are deleted and the `screen` rules are mutated to `all` 
    - [x] Merge each exported slide PDF buffer into a single PDF document:
        - In-memory merging of individual slide PDF export is achieved via [HummusJS](https://github.com/galkahana/HummusJS)

2. Test supported presentation frameworks:
    - [x] Bespoke.js
    - [x] CSSS
    - [x] deck.js
    - [x] DZSlides
    - [x] Flowtime.js
    - [x] impress.js
    - [x] Remark
    - [x] Reveal.js
    - [x] RISE
    - [x] Shower
    - [x] Slidy
    - [x] WebSlides
    - [x] Generic plugin

3. Misc:
    - [x] Screenshots
    - [x] Update README
    - [ ] Standalone executables
    - [x] Docker image

**Comment by astefanutti** [comment]
Initial implementation has started in the [puppeteer](https://github.com/astefanutti/decktape/tree/puppeteer) branch.

In-memory merging of individual slide PDF export is achieved via [HummusJS](https://github.com/galkahana/HummusJS), which seems to work well. The only drawback is that we lost reusable graphics for the time being hence an increase in final PDF size. However it may be possible to achieve reuse with HummusJS.

**Comment by eljefe6a** [comment]
I wrote a [PDF output](https://github.com/eljefe6a/RemarkMacros/blob/master/remarkprinter.js) for Remark. This method may not work with other presentation projects, but Remark does recognize the print mode.

One issue I found with this was that some text didn't show up. I haven't figured out that was happening. It only happened on code sections (this didn't happen when printing from GUI). You might hit this too.

**Comment by astefanutti** [comment]
@eljefe6a thanks for the heads-up. Indeed Remark, as well as Shower, seem to work pretty well with Chrome PDF print. That being said, I'd be inclined to keep the Decktape principle of _screen printing_ even for these frameworks. That would offer an alternative and keep the code simpler.

I've been able to export the _standard_ Remark presentation (based on my current work with headless Chrome and Puppeteer) and haven't noticed any issue. Maybe that's something specific to your presentation?

**Comment by eljefe6a** [comment]
It took me a few hours, but I figured out that issue. There seem to be some differences between running Chrome in headless versus the GUI.

This one turned out to be a web font issue. When I installed the font as system font, it worked with headless. The web font rendered fine when running in regular Chrome. To see if it was an issue with spinning up Chrome, I disabled headless and rendered the PDF manually. That worked.

When I was debugging this issue, I compared the files. With headless, the PDF header outputs as `Chromium`. In the GUI, the PDF header outputs as `Chrome`.
